### PR TITLE
Implement FixIt for C#

### PR DESCRIPTION
**Pull request by mispencer**
Implement https://github.com/Valloric/ycmd/pull/167 for C#.

I have signed the CLA.

### CODE REVIEW

File: ycmd/tests/testdata/testy/FixItTestCase.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+using System;
+using System.Text;
+
+namespace testy {
+	public class GetTypeTestCase {
+		public GetTypeTestCase() {
+			var str = "";
+			str.EndsWith("A");
+			var i = 5;
+			var j = i + 5;
```
Comment (Valloric) [inline ycmd/tests/testdata/testy/FixItTestCase.cs]: Consider adding a comment in the file describing the expected fix-it.

**Comment by Valloric** [comment]
This looks nice, thanks for sending it in!

@puremourning I'm interested in your thoughts on this implementation since you implemented the libclang support.

**Comment by mispencer** [comment]
I updated the pull request to resolve issues if there wasn't any text changes or the text change affected the first or last character in the buffer. I also added some tests.

Regarding comments in the test data file: I would rather not, since comments are not negligible here - they affect the fix output.

**Comment by Valloric** [comment]
> Regarding comments in the test data file: I would rather not, since comments are not negligible here - they affect the fix output.

Oh, that makes sense.

**Comment by puremourning** [comment]
@Valloric this looks ok to me. I've tested with the client changes and incorporated them into PR https://github.com/Valloric/YouCompleteMe/pull/1555

**Comment by Valloric** [comment]
Thanks you both!

@homu r+

**Pull request merged**